packages/core/src/page.js

```diff
@@ -18,6 +18,7 @@ export class Page {
     this.session = session;
     this.browser = session.browser;
     this.enableJavaScript = options.enableJavaScript ?? true;
+    this.width = null;
```

Comment:
dont pass this around everywhere - just update it in `intercept` like `currentWidth` and set it when you resize the page.